pkg/kubectl/cmd/create_role.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+var (
+	roleLong = templates.LongDesc(`
+		Create a Role to hold a logic grouping of permissions.`)
+
+	roleExample = templates.Examples(`
+		# Create a Role named "pod-reader" that allows user to perform "get", "watch" and "list" on "pods"
+		kubectl create role pod-reader --verb=get --verb=list --verb=watch --resource=pods`)
+)
+
+// Role is a command to ease creating Roles.
+func NewCmdCreateRole(f cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "role NAME --verb=verb --resource=resource [--api-group=apigroup] [--resource-name=resourcename] [--non-resource-url=nonresourceurl] [--dry-run]",
```

Comment:
When you run `kubectl` commands, you can actually do `kubect get resource.group`.  We could wire this command up to do the same thing so that I can do something like this: `kubectl create role foo --verb=get,list,watch --resource=jobs.batch,deployments.extensions,rs.extensions` (note the use of the shortcut) and come out the other end with a working role.